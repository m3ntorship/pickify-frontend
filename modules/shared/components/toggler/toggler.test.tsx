import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Toggler from './index';

describe('Toggler Component with snapshot', () => {
  it('should render classnames include (w-7 h-4) and (opacity-25) when passing disabled={true} and size=sm ', () => {
    const tree = renderer
      .create(
        <Toggler
          size="sm"
          disabled
          isChecked={false}
          handleTogglerClick={(): boolean => {
            return true;
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render classnames include (w-7 h-4) and do NOT include (opacity-25) when passing disabled={false} and size=sm ', () => {
    const tree = renderer
      .create(
        <Toggler
          size="sm"
          disabled={false}
          isChecked={false}
          handleTogglerClick={(): boolean => {
            return true;
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render include (w-sm h-6) and classnames include (opacity-25) when passing disabled={true} and size=default ', () => {
    const tree = renderer
      .create(
        <Toggler
          size="default"
          disabled
          isChecked={false}
          handleTogglerClick={(): boolean => {
            return true;
          }}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render include (w-sm h-6) and classnames do NOT include (opacity-25) when passing disabled={false} and size=default ', () => {
    const tree = render(
      <Toggler
        size="default"
        disabled={false}
        isChecked={false}
        handleTogglerClick={(): boolean => {
          return true;
        }}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
describe('Toggler Component with react testing library', () => {
  it('should be clicked once', () => {
    const handleClick = jest.fn();
    render(
      <Toggler
        size="sm"
        disabled={false}
        isChecked={false}
        handleTogglerClick={handleClick}
      />,
    );
    const calledonce = 1;
    userEvent.click(screen.getByTestId('toggler-test'));
    expect(handleClick).toHaveBeenCalledTimes(calledonce);
  });
});
