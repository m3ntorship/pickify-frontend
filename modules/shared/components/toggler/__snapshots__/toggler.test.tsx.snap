// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggler Component with snapshot should render classnames include (w-7 h-4) and (opacity-25) when passing disabled={true} and size=sm  1`] = `
<label
  className="rounded-full flex items-center cursor-pointer transition duration-100 ease-out focus:outline-none relative w-7 h-4  bg-grey-shd4 hover:bg-grey-shd3  focus-within:ring-1 focus-within:ring-grey-shd3 cursor-not-allowed opacity-25"
  htmlFor="test"
>
  <input
    className="opacity-0 absolute cursor-not-allowed"
    data-testid="test"
    disabled={true}
    id="test"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="rounded-full transition-all duration-100 ease-in-out bg-white w-2.5 h-2.5 m-2xxs"
  />
</label>
`;

exports[`Toggler Component with snapshot should render classnames include (w-7 h-4) and do NOT include (opacity-25) when passing disabled={false} and size=sm  1`] = `
<label
  className="rounded-full flex items-center cursor-pointer transition duration-100 ease-out focus:outline-none relative w-7 h-4  bg-grey-shd4 hover:bg-grey-shd3  focus-within:ring-1 focus-within:ring-grey-shd3"
  htmlFor="test"
>
  <input
    className="opacity-0 absolute cursor-pointer"
    data-testid="test"
    disabled={false}
    id="test"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="rounded-full transition-all duration-100 ease-in-out bg-white w-2.5 h-2.5 m-2xxs"
  />
</label>
`;

exports[`Toggler Component with snapshot should render include (w-sm h-6) and classnames do NOT include (opacity-25) when passing disabled={false} and size=default  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="rounded-full flex items-center cursor-pointer transition duration-100 ease-out focus:outline-none relative w-2xl h-6  bg-grey-shd4 hover:bg-grey-shd3  focus-within:ring-2 focus-within:ring-grey-shd3"
        for="test"
      >
        <input
          class="opacity-0 absolute cursor-pointer"
          data-testid="test"
          id="test"
          type="checkbox"
        />
        <div
          class="rounded-full transition-all duration-100 ease-in-out bg-white w-4 h-4 m-1 "
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="rounded-full flex items-center cursor-pointer transition duration-100 ease-out focus:outline-none relative w-2xl h-6  bg-grey-shd4 hover:bg-grey-shd3  focus-within:ring-2 focus-within:ring-grey-shd3"
      for="test"
    >
      <input
        class="opacity-0 absolute cursor-pointer"
        data-testid="test"
        id="test"
        type="checkbox"
      />
      <div
        class="rounded-full transition-all duration-100 ease-in-out bg-white w-4 h-4 m-1 "
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toggler Component with snapshot should render include (w-sm h-6) and classnames include (opacity-25) when passing disabled={true} and size=default  1`] = `
<label
  className="rounded-full flex items-center cursor-pointer transition duration-100 ease-out focus:outline-none relative w-2xl h-6  bg-grey-shd4 hover:bg-grey-shd3  focus-within:ring-2 focus-within:ring-grey-shd3 cursor-not-allowed opacity-25"
  htmlFor="test"
>
  <input
    className="opacity-0 absolute cursor-not-allowed"
    data-testid="test"
    disabled={true}
    id="test"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    className="rounded-full transition-all duration-100 ease-in-out bg-white w-4 h-4 m-1 "
  />
</label>
`;
